r/R/util.R

```diff
@@ -251,3 +251,25 @@ augment_io_error_msg <- function(e, call, schema = NULL, format = NULL) {
   handle_augmented_field_misuse(msg, call)
   abort(msg, call = call)
 }
+
+simulate_data_frame <- function(schema) {
+
+  arrays <- lapply(
+    schema$fields,
+    function(field) tryCatch(
+      concat_arrays(type = field$type),
+      error = function(...) concat_arrays(type = NULL)
```

Comment:
Why would this error ever occur?

Reply:
Something went horribly sideways with tests for extension types as `concat_arrays()` casts from a NULL to the given type, but this doesn't work with extension types. Does that sound like a bug to you?